type_description_interfaces/srv/GetTypeDescription.srv

```diff
@@ -0,0 +1,27 @@
+# ROS interface type name, in PACKAGE/NAMESPACE/TYPENAME format
+string type_name
+
+# REP-2011 RIHS hash string, which looks like RIHSxx_hashvalue
+string type_version_hash
+---
+# True if the type description information is available and populated in the response
+bool successful
+# Empty if 'successful' was true, otherwise contains details on why it failed
+string failure_reason
+
+# The idl or msg file name
+string type_description_raw_file_name
+
+# The idl or msg file, with comments and whitespace
+# The file extension and/or the contents can be used to determine the format
+# This contains all recursive raw files, each separated by a line of 80 '=' characters
+# The order will be the same as the TypeDescription -
+# which is top level interface then referenced type descriptions in alphabetical order by type_name
+string type_description_raw
```

Comment:
I'm not so sure I like shoving all of this in the response. I would prefer this service return _just_ the type description, and that we make the rest of the information available through other calls.

What does this contain when the dynamic types API is used to construct a type on the fly?

Reply:
What about switchable via a `bool` in the request? That way we don't have to define two interfaces, but we can switch on or off the overhead of including all the raw text.

The raw text would have to be empty for a dynamic type, I assume. There is no "source". Maybe there's a bool in the response stating whether it's a system-loaded type or a dynamic one?

Reply:
We discussed this in the REP and decided to not have a flag to control this behavior. We can reevaluate that decision, but we did discuss/decide on it already.

Reply:
What about the question of, what do we put in `type_description_raw` for dynamic types?